[#1232] update readme

boggledekoog opened the pull request:
[#1232] updated readme to contain all operators and modules. Also added link to the wiki and xml syntax highlighting

The list of operators is quite long now, we could consider moving it into the wiki.

ChrizZz110 commented on README.md:
```diff
@@ -169,6 +198,11 @@ The main contents of that module are the EPGM data model and a corresponding POJ
 
+### gradoop-data-integration
+
+This module contains graph importers (e.g from CSV files) and transformations on graphs (e.g
```
```
Provides functionalities to support graph data integration. This includes minimal CSV and JSON importers as well as graph transformation operators (e.g. connect neighbors or conversion of edges to vertices and vice versa).
```

ChrizZz110 commented on README.md:
```diff
@@ -54,13 +58,37 @@ The following tables contains an overview (GC = Graph Collection, G = Logical Gr
+| Operator                    | Output | Output description                                           | Impl |
+|:----------------------------|:-------|:-------------------------------------------------------------|:----:|
+| Aggregation                 | G      | Graph with result of an aggregate function as a new property | Yes  |
+| Matching                    | GC     | Graphs that match a given graph pattern                      | Yes  |
```
I would split Matching in "Cypher" and "Query" to retain the naming of the operators itself.
Cypher: Graphs that match a given Cypher graph pattern
Query: Graphs that match a given Cypher-based GDL graph pattern

ChrizZz110 commented on README.md:
```diff
@@ -54,13 +58,37 @@ The following tables contains an overview (GC = Graph Collection, G = Logical Gr
+| ConnectNeighbors            | G      | Connects neighbors of a vertex.                              | Yes  |
+| EdgeToVertex                | G      | Converts an edge to a vertex.                                | Yes  |
+| VertexToEdge                | G      | Converts a vertex to an edge.                                | Yes  |
+| RollUp                      | GC     | Generates combinations according to the rollup operation in SQL. | Yes  |
```
I would move RollUp below "Grouping" because it is related. Change output description to: Graphs representig structural condense of the input graph according to SQL RollUp

ChrizZz110 commented on README.md:
```diff
@@ -54,13 +58,37 @@ The following tables contains an overview (GC = Graph Collection, G = Logical Gr
+| SingleSourceShortestPaths   | G      | Operator wrapping SingleSourceShortestPaths integrated in Flink&trade; Gelly. | Yes  |
+| TriangleCounting            | G      | Operator wrapping TriangleCounting integrated in Flink&trade; Gelly. | Yes  |
+| DistinctVertexDegrees       | G      | Operator VertexDegrees calculation integrated in Flink&trade; Gelly. | Yes  |
+| PropertyExtraction          | G      | Extracts a property of a vertex.                             | Yes  |
```
```suggestion
| PropertyExtraction          | G      | Extracts a property of a vertex to a new vertex.                             | Yes  |
```

ChrizZz110 commented on README.md:
```diff
@@ -54,13 +58,37 @@ The following tables contains an overview (GC = Graph Collection, G = Logical Gr
+| Grouping                    | G      | Structural condense of the input graph                       | Yes  |
+| Subgraph                    | G      | Subgraph that fulfils given vertex and edge predicates       | Yes  |
+| Cloning                     | G      | Creates a copy of a logical graph.                           | Yes  |
+| PageRankSampling            | G      | Creates a PageRank-Sampling of a logical graph.              | Yes  |
```
I would condens all samplings to a single "Sampling" operator.

ChrizZz110 commented on README.md:
```diff
@@ -54,13 +58,37 @@ The following tables contains an overview (GC = Graph Collection, G = Logical Gr
+| TriangleCounting            | G      | Operator wrapping TriangleCounting integrated in Flink&trade; Gelly. | Yes  |
+| DistinctVertexDegrees       | G      | Operator VertexDegrees calculation integrated in Flink&trade; Gelly. | Yes  |
+| PropertyExtraction          | G      | Extracts a property of a vertex.                             | Yes  |
+| ConnectNeighbors            | G      | Connects neighbors of a vertex.                              | Yes  |
```
```suggestion
| ConnectNeighbors            | G      | Connects neighbors of a vertex with newly created edges.                              | Yes  |
```

ChrizZz110 commented on README.md:
```diff
@@ -54,13 +58,37 @@ The following tables contains an overview (GC = Graph Collection, G = Logical Gr
+| VertexToEdge                | G      | Converts a vertex to an edge.                                | Yes  |
+| RollUp                      | GC     | Generates combinations according to the rollup operation in SQL. | Yes  |
+| Split                       | GC     | Splits a logical graph into a GraphCollection. | Yes  |
+| BusinessTransactionGraphs   | GC     | EPGMVertex-centric implementation to isolate business transaction graphs. | Yes  |
```
This can be deleted.

ChrizZz110 commented on README.md:
```diff
@@ -169,6 +198,11 @@ The main contents of that module are the EPGM data model and a corresponding POJ
 implementation which is used in Flink&trade;. The persistent representation of the EPGM
```
```suggestion
implementation which is used in Flink&reg;. The persistent representation of the EPGM
```

boggledekoog commented:
The ticket can be reviewed again, I integrated all proposed changes.

I moved the list of operators from the README to the [Wiki](https://github.com/dbs-leipzig/gradoop/wiki/List-of-Operators) and added links to further operator documentation.
Also the Split operator was missing from the list.

ChrizZz110 commented on README.md:
```diff
@@ -50,54 +54,7 @@ properties even if they have the same label.
-| Apply         | GC   | GC   | Applies unary operator (e.g. aggregate) on each graph in the collection | Yes  |
-| Reduce        | GC   | G    | Reduces collection to single graph using binary operator (e.g. combine) | Yes  |
-| Call          | GC/G | GC/G | Applies external algorithm on graph or graph collection                 | Yes  |
+An overview of the implemented operators can be found [here](https://github.com/dbs-leipzig/gradoop/wiki/List-of-Operators).
```
```suggestion
An overview and detailed descriptions of the implemented operators can be found in the [Gradoop Wiki](https://github.com/dbs-leipzig/gradoop/wiki/List-of-Operators).
```
